### PR TITLE
chore: improve cmd2 maintenance path

### PR DESCRIPTION
**Pull request by lphuc2250gma**
Summary:
- Add focused edge-case unit tests in tests/test_utils.py for small pure helpers in cmd2/utils.py (e.g., quote/unquote, alphabetical_sort, str_to_bool, strip_ansi handling of empty/None/whitespace inputs and mixed-quote tokens), and tighten one or two narrow type annotations where Optional/Union is currently broader than the implementation requires.
- Keep the change narrow so it is straightforward to review.

Notes:
- I kept this scoped to the relevant implementation and tests.

### CODE REVIEW

File: cmd2/utils.py [hunk 1]
```diff
@@ -415,7 +414,7 @@ class StdSim:
 
     def __init__(
         self,
-        inner_stream: Union[TextIO, "StdSim"],
+        inner_stream: TextIO | "StdSim",
```
Comment (tleonhardt) [inline cmd2/utils.py:417]: For the upcoming `cmd2` 4.0.0 release we support Python versions 3.11 or newer.

Using the `|` operator with a forward-declaration string type works with Python 3.14+, but does not work with older versions of Python, hence the `Union` is still required for forward declaration types.

**Review by tleonhardt — CHANGES_REQUESTED** [review]

**Comment by kmvanbrunt** [comment]
Closing this AI-generated PR.

File: tests/test_utils.py [hunk 2]
```diff
@@ -285,6 +317,24 @@ def test_to_bool_float() -> None:
...
+    with pytest.raises(ValueError):
+        cu.to_bool("\t")
+    with pytest.raises(ValueError):
+        cu.to_bool("\n")
+
+
+def test_to_bool_none() -> None:
+    assert not cu.to_bool(None)
+
+
```
Comment (anselor) [inline tests/test_utils.py:337]: All of this could've been a loop over a list of checks.
